components/org.wso2.carbon.identity.conditional.auth.functions.analytics/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/analytics/CallAnalyticsFunctionImpl.java

```diff
@@ -60,8 +62,11 @@
     private static final String PARAM_INPUT_STREAM = "InputStream";
 
     @Override
-    public void callAnalytics(Map<String, String> metadata,
-                              Map<String, Object> payloadData, Map<String, Object> eventHandlers) {
+    public void callAnalytics(Map<String, String> metadataJsObject,
+                              Map<String, Object> payloadDataJsObject, Map<String, Object> eventHandlers) {
+
+        Map<String, String> metadata = new HashMap<>(metadataJsObject);
+        Map<String, Object> payloadData = new HashMap<>(payloadDataJsObject);
```

Comment:
What is the requirement for this initialization?

Reply:
With Graal Execution original `metadataJsObject` and `payloadDataJsObject` have the scope of their `Context` and become unavailable in the when AsyncProcess` is finally executed and the `Context` would have been closed